plugins/NebulaTextures/src/gui/NebulaTexturesDialog.cpp

```diff
@@ -0,0 +1,1419 @@
+		setTexturesVisible(CUSTOM_TEXNAME,false);
+		setTexturesVisible(DEFAULT_TEXNAME,true);
+	}
+	else{
+		setTexturesVisible(CUSTOM_TEXNAME,true);
+		avoidConflict(); // only when show_custom && avoid_conflict
+	}
+}
+
+
+/*
+ * Set the visibility of the specified texture and its sub-tiles.
+ *
+ * Steps:
+ *   - Retrieve the texture identified by `TexName`.
+ *   - If the texture is found, iterate through its sub-tiles.
+ *   - Set the visibility of each sub-tile based on the `visible` parameter.
+ *   - Return false if the texture or its sub-tiles cannot be found.
+ *
+ * @param TexName  The name of the texture to modify.
+ * @param visible  The visibility state to set for the texture and its sub-tiles.
+ *
+ * @return true if the texture and its sub-tiles are found and visibility is set successfully.
+ *         false if the texture or its sub-tiles cannot be found.
+ */
+bool NebulaTexturesDialog::setTexturesVisible(QString TexName, bool visible)
+{
+	StelSkyImageTile* mTile = get_aTile(TexName);
+	if(!mTile) return false;
+	for (int i = 0; i < mTile->getSubTiles().size(); ++i) {
+		auto subTile = mTile->getSubTiles()[i];
+		StelSkyImageTile* subImageTile = dynamic_cast<StelSkyImageTile*>(subTile);
+		subImageTile->setVisible(visible);
+	}
+	return true;
+}
+
+
+/*
+ * Retrieve a StelSkyImageTile object based on the provided texture key.
+ *
+ * Searches for the sky layer associated with the given key in the StelSkyLayerMgr.
+ * If found, it attempts to cast the layer to a StelSkyImageTile and returns it.
+ * Returns Q_NULLPTR if no layer is found or the cast fails.
+ */
+StelSkyImageTile* NebulaTexturesDialog::get_aTile(QString key)
+{
+	StelSkyLayerMgr* skyLayerMgr = GETSTELMODULE(StelSkyLayerMgr);
+	// Keep the same form with getShortName()
+	QString keyname = qc_(key, "dataset short name");
+	auto aTex = skyLayerMgr->allSkyLayers.find(keyname);
+	if(aTex == skyLayerMgr->allSkyLayers.end())
+		return Q_NULLPTR;
+	StelSkyLayerMgr::SkyLayerElem* aElem = aTex.value();
+	if (!aElem || !aElem->layer) return Q_NULLPTR;
+	StelSkyImageTile* aTile = dynamic_cast<StelSkyImageTile*>(aElem->layer.data());
+	return aTile;
+}
+
+
+/*
+ * Avoid conflicts between custom and default textures by hiding overlapping tiles.
+ *
+ * Steps:
+ *   - Check if the 'Avoid Area Conflict' flag is enabled.
+ *   - Retrieve the custom and default texture tiles.
+ *   - Loop through the sub-tiles of the default texture.
+ *   - For each visible sub-tile, check for overlap with custom texture sub-tiles.
+ *   - Hide the default texture sub-tile if overlap is detected.
+ *   - If conflicts are not to be avoided, make all default texture tiles visible.
+ */
+void NebulaTexturesDialog::avoidConflict()
+{
+	if (getAvoidAreaConflict()){
+		StelSkyImageTile* cusTile = get_aTile(CUSTOM_TEXNAME);
+		StelSkyImageTile* defTile = get_aTile(DEFAULT_TEXNAME);
+		if(!cusTile) return;
+		if(!defTile) return;
+
+		for (int i = 0; i < defTile->getSubTiles().size(); ++i) {
```

Comment:
Here and elsewhere, please adhere to our coding style for braces, these usually go on new lines. 

In the else clause below,  if it's just one statement, it can go without braces.